geom/wkb_marshal.go

```diff
@@ -87,10 +86,5 @@ func (m *wkbMarshaler) writeSequence(seq Sequence) {
 // floatsAsBytes reinterprets the floats slice as a bytes slice in a similar
 // manner to reinterpret_cast in C++.
 func floatsAsBytes(floats []float64) []byte {
-	var byts []byte
-	hdr := (*reflect.SliceHeader)(unsafe.Pointer(&byts))
-	hdr.Data = (*reflect.SliceHeader)(unsafe.Pointer(&floats)).Data
-	hdr.Len = 8 * len(floats)
-	hdr.Cap = 8 * cap(floats)
-	return byts
+	return unsafe.Slice((*byte)(unsafe.Pointer(&floats[0])), 8*len(floats))
```

Comment:
We need to be careful about the case where `len(floats)` is `0`, otherwise `floats[0]` will panic. We could just have a guard if statement checking that condition first (and simply return `nil` in that case). That applies to `bytesAsBytes` too.